mountpoint-s3-fs/src/superblock/readdir.rs

```diff
@@ -98,6 +99,7 @@ impl ReaddirHandle {
             }
         };
 
+        #[cfg(feature = "manifest")]
```

Comment:
The code duplication here is not ideal. We should consider refactoring to avoid it. E.g. by extracting `iter` creation into its own function.

Could be a follow up, though.